### PR TITLE
Add support for ALTER TABLE DROP PARTITION

### LINKED ISSUE #1059: Hive Dialect for 'drop partition' error
sqlglot cannot parse 'Alter table drop partition'
image
image

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #1059

### CODE REVIEW

File: sqlglot/generator.py [hunk 1]
```diff
@@ -629,13 +634,7 @@ def national_sql(self, expression: exp.National) -> str:
...
 
     def partition_sql(self, expression: exp.Partition) -> str:
-        keys = csv(
-            *[
-                f"{prop.name}={self.sql(prop, 'value')}" if prop.args.get("value") else prop.name
-                for prop in expression.this
-            ]
-        )
-        return f"PARTITION({keys})"
+        return f"PARTITION({self.expressions(expression, 'this')})"
```
Comment (tobymao) [inline sqlglot/generator.py:637]: in generally expressions should be called with "expressions", not with "this"

File: sqlglot/generator.py [hunk 2]
```diff
@@ -1419,14 +1418,19 @@ def altertable_sql(self, expression: exp.AlterTable) -> str:
             actions = self.expressions(expression, "actions")
         elif isinstance(actions[0], exp.Delete):
             actions = self.expressions(expression, "actions", flat=True)
-        elif isinstance(actions[0], self.WITH_SINGLE_ALTER_TABLE_ACTION):
+        elif actions[0].__class__ in self.WITH_SINGLE_ALTER_TABLE_ACTION:
```
Comment (tobymao) [inline sqlglot/generator.py:1421]: why was this needed

Comment (GeorgeSittas) [reply sqlglot/generator.py:1421]: Not necessary to use, it's only helpful for reporting "unsupported" errors I think. We can probably put all cases under the `else` branch instead. Should I do that?

**Review by tobymao — APPROVED** [review]
